packages/backend/src/open_payments/payment/outgoing/service.ts

```diff
@@ -59,61 +60,64 @@ export async function createOutgoingPaymentService(
     logger: deps_.logger.child({ service: 'OutgoingPaymentService' })
   }
   return {
-    get: (id) => getOutgoingPayment(deps, id),
+    get: (id, clientId) => getOutgoingPayment(deps, id, clientId),
     create: (options: CreateOutgoingPaymentOptions) =>
       createOutgoingPayment(deps, options),
     fund: (options) => fundPayment(deps, options),
     processNext: () => worker.processPendingPayment(deps),
-    getPaymentPointerPage: (paymentPointerId, pagination) =>
-      getPaymentPointerPage(deps, paymentPointerId, pagination)
+    getPaymentPointerPage: (paymentPointerId, pagination, clientId) =>
+      getPaymentPointerPage(deps, paymentPointerId, pagination, clientId)
   }
 }
 
 async function getOutgoingPayment(
   deps: ServiceDependencies,
-  id: string
+  id: string,
+  clientId?: string
 ): Promise<OutgoingPayment | undefined> {
-  const outgoingPayment = await OutgoingPayment.query(deps.knex)
-    .findById(id)
-    .withGraphJoined('quote.asset')
+  let outgoingPayment: OutgoingPayment
+  if (!clientId) {
+    outgoingPayment = await OutgoingPayment.query(deps.knex)
+      .findById(id)
+      .withGraphJoined('quote.asset')
+  } else {
+    outgoingPayment = await OutgoingPayment.query(deps.knex)
+      .findById(id)
+      .withGraphJoined('[quote.asset, grant]')
+      .where('grant.clientId', clientId)
+  }
   if (outgoingPayment) return await addSentAmount(deps, outgoingPayment)
   else return
 }
 
 export interface CreateOutgoingPaymentOptions {
   paymentPointerId: string
   quoteId: string
+  grant?: Grant | string
```

Comment:
Can this still just be
```suggestion
  grant?: Grant
```

Reply:
If we change that, should incoming payments then also not have `grantId` but `grant`?

Reply:
I think it's ok for `CreateOutgoingPaymentOptions` to take the entire `grant` and `CreateIncomingPaymentOptions` to have `grantId` because they have differnt needs. Outgoing payments needs to validate grant access limits.